ADAPIService/implementations/ADService.cs

```diff
@@ -62,16 +65,76 @@ public class OperationResult
 
     public IActionResult ModifyADObject(ModifyModel newAdObject, string adObjectType)
     {
+        using DirectorySearcher searcher = new DirectorySearcher();
+        searcher.Filter = $"({newAdObject.Identifier?.Attribute}={newAdObject.Identifier?.Value})";
+        SearchResult result = searcher.FindOne();
+
         try
         {
-            _repository.ModifyADObject(newAdObject, adObjectType);
+            DirectoryEntry objectEntry = result.GetDirectoryEntry();
+            _repository.ModifyADObject(newAdObject, adObjectType, objectEntry);
+
+            foreach (var action in newAdObject.Actions)
+            {
+                if (action.Value is not null)
+                {
+                    HandleAction(action.Key, action.Value, objectEntry);
+                }
+            }
             return new OkObjectResult($"{newAdObject} modify successfully");
         }
         catch (Exception ex)
         {
             _logger.LogError(ex, $"An error occurred while modify {newAdObject}, error message:", ex.Message);
             return new StatusCodeResult(500);
         }
+        
+    }
+
+    public void HandleAction(string actionKey, Dictionary<string, object> actionValue, DirectoryEntry objectEntry)
+    {
+        MethodInfo method = typeof(Repository).GetMethod(actionKey);
```

Comment:
Security breach,
If I type invoke as method name I can invoke methods from repository and execute code on your server. Limit your methods